src/userSync.js

```diff
@@ -315,7 +315,9 @@ export function newUserSync(userSyncDependencies) {
   return publicApi;
 }
 
-const browserSupportsCookies = !isSafariBrowser() && storage.cookiesAreEnabled();
+const browserSupportsCookies = function() {
```

Comment:
Does the type of this variable really need to change to a function, instead of its boolean resolution?

If so, can you please update the corresponding unit tests in `test/spec/userSync_spec.js` to update the signature references it has for this param (from a boolean to a function)?

Reply:
This way the check is done later. At the time it is done now, the cmp(-module?) isn't ready, or at least `gdprDataHandler.getConsentData()` doesn't deliver anything. When called at time needed, the module or the cmp is set and we get response.

I'll update the PR